### PR TITLE
fix/cfd 558 circular return

### PR DESCRIPTION
**Pull request by Aaron-Collis-IW**
# Description

Create a validation error which alerts the user that they can't use a one direction journey for return and point to point period products. 

# Testing instructions

-   On test:
- Use noc FRSV
- Try and create a p2p period and return for the 1a service

# Type of change

-   [x] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

### CODE REVIEW

File: repos/fdbt-site/src/pages/api/service.ts [hunk 1]
```diff
@@ -17,8 +17,27 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
...
 
         const service = await getServiceByIdAndDataSource(getAndValidateNoc(req, res), serviceId, dataSource);
+        const directions = Array.from(
+            service.journeyPatterns.reduce((set, pattern) => {
+                set.add(pattern.direction);
+                return set;
+            }, new Set<string>()),
+        );
+
+        if (directions.length === 1 && isReturn) {
```
Comment (RobCatton) [inline repos/fdbt-site/src/pages/api/service.ts:31]: What will happen here if we have outbound + clockwise (or a pairing which both mean the same thing)? That would mean there are multiple directions in this array, but they both mean the same thing.

File: repos/fdbt-site/src/pages/api/service.ts [hunk 2]
```diff
@@ -17,8 +17,27 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
...
+            const errors: ErrorInfo[] = [
+                {
+                    id: 'service',
```
Comment (RobCatton) [inline repos/fdbt-site/src/pages/api/service.ts:34]: Do we want to pass back the selected service to use as the default selection? The error message mentions a service, but it wont be shown on the screen.

File: repos/fdbt-site/src/pages/api/service.ts [hunk 3]
```diff
@@ -17,8 +17,27 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
...
+                    errorMessage: 'The chosen a service only has one direction and can`t be used for a return',
```
Comment (RobCatton) [inline repos/fdbt-site/src/pages/api/service.ts:35]: Error message needs slightly altering, the third word seems unnecessary.

File: repos/fdbt-site/src/pages/api/service.ts (hunk 1 above)
Comment (jacsamell) [reply repos/fdbt-site/src/pages/api/service.ts:31]: Yeah good point, we need to check that they have inbound + outbound below 👍

File: repos/fdbt-site/src/pages/api/service.ts (hunk 3 above)
Comment (Aaron-Collis-IW) [reply repos/fdbt-site/src/pages/api/service.ts]: done

File: repos/fdbt-site/src/pages/api/service.ts (hunk 1 above)
Comment (Aaron-Collis-IW) [reply repos/fdbt-site/src/pages/api/service.ts]: this is done

File: repos/fdbt-site/src/pages/api/service.ts (hunk 2 above)
Comment (Aaron-Collis-IW) [reply repos/fdbt-site/src/pages/api/service.ts:36]: this is done

**Pull request merged**